File: packages/ts/src/has-bare-name.ts
import { isBareName } from "./is-bare-name.js";
import type { Toml } from "./load-extension-entries.js";

/** True when some entry's `path` names an extension by bare package name. */
export function hasBareName(entries: Toml[]): boolean {
  return entries.some((e) => typeof e.path === "string" && isBareName(e.path));
}

File: packages/python/dirsql/has_bare_name.py
"""Whether any of a config's extension entries names a package."""

from __future__ import annotations

from .is_bare_name import is_bare_name


def _has_bare_name(entries):
    return any(
        isinstance(e, dict)
        and isinstance(e.get("path"), str)
        and is_bare_name(e["path"])
        for e in entries
    )
